src/game_events/handlers.cpp

```diff
@@ -41,13 +52,49 @@ namespace game_events
 {
 /* ** event_handler ** */
 
-event_handler::event_handler(config&& cfg, bool imi, const std::vector<std::string>& types)
-	: first_time_only_(cfg["first_time_only"].to_bool(true))
-	, is_menu_item_(imi)
+event_handler::event_handler(const std::string& types, const std::string& id)
+	: first_time_only_(true)
+	, is_menu_item_(false)
 	, disabled_(false)
-	, cfg_(cfg)
+	, is_lua_(false)
 	, types_(types)
+{}
+
+std::vector<std::string> event_handler::names(const variable_set* vars) const
+{
+	std::string names = types_;
+	if(vars && utils::might_contain_variables(names)) {
+		names = utils::interpolate_variables_into_string(names, *vars);
+	}
+
+	// Do some standardization on the name field.
+	// Split the name field...
+	std::vector<std::string> standardized_names = utils::split(names);
+
+	if(!names.empty()) {
+		// ...and standardize each one individually. This ensures they're all valid for by-name lookup.
+		for(std::string& single_name : standardized_names) {
+			if(utils::might_contain_variables(single_name) && !vars) {
+				// If we don't have gamedata, we can't interpolate variables, so there's
+				// no way the name will match. Move on to the next one in that case.
+				single_name.clear();
+			} else {
+				single_name = event_handlers::standardize_name(single_name);
+			}
+		}
+
+		assert(!standardized_names.empty());
```

Comment:
How do we know that UMC can't trigger this assert? It looks as if it might trigger when there's a `$` in a name.

Reply:
This code was copy-pasted from elsewhere and then altered, so it's possible that this assertion just isn't correct anymore.